[BUG]: Build fails on GCC 16 due to -Werror (unused-but-set-variable, format-truncation)

### Tvheadend Version
v4.3 (stable)

### Build Number
v4.3-2681-gbb09b4fae

### Platform
Other.. (Fedora 44)

### Architecture
ARMv8 (aarch64)

### Category
Packaging

### Description
Building current master (bb09b4fae) on Fedora 44 with GCC 16.1.1 fails. Two
long-standing constructs are now rejected under the project's default `-Werror`:

1. `src/ratinglabels.c:615` — a local variable that is set but never read:

```
src/ratinglabels.c:615:7: error: variable 'a' set but not used [-Werror=unused-but-set-variable=]
  615 |   int a=1;
      |       ^
```

2. `src/input/mpegts/satip/satip_rtsp.c:364` — `_w[16]` is too small to hold
   `"&tvhweight=%d"` for the full range of `int`:

```
src/input/mpegts/satip/satip_rtsp.c:364:42: error: '%d' directive output may be truncated writing between 1 and 10 bytes into a region of size 5 [-Werror=format-truncation=]
```

Both are dormant (the satip line dates to 2019, the ratinglabels line to 2023);
GCC 16's stricter diagnostics newly promote them to errors.

This isn't caught in CI because the `build-rpm-native` (Fedora) job is marked
`continue-on-error: true`, so its failures don't fail the pipeline.

Fix: #2136

### Reproduction
1. `../configure && make` on Fedora 44 (GCC 16.1.1)
2. Build aborts at `src/ratinglabels.o` / `satip_rtsp.o`.

--- a/src/input/mpegts/satip/satip_rtsp.c
+++ b/src/input/mpegts/satip/satip_rtsp.c
@@ -340,11 +340,12 @@ satip_rtsp_play( http_client_t *hc, const char *pids,
                  int max_pids_len, int weight )
 {
   char *stream = NULL;
-  char _stream[32], _w[16];
+  char _stream[32], _w[24];
   const char *p[8], *add[8], *del[8];
   int pcnt, addcnt, delcnt;
   int i, r = 0, index = 0;
-  char buf[max_pids_len + 32];
+  /* room for "delpids=%s&addpids=%s%s": del+add up to max_pids_len, plus _w */
+  char buf[(max_pids_len < 32 ? 32 : max_pids_len) + 64];
 
   if (max_pids_len < 32)
     max_pids_len = 32;

--- a/src/ratinglabels.c
+++ b/src/ratinglabels.c
@@ -612,6 +612,8 @@ ratinglabel_class_get_icon ( void *obj )
 static void
 ratinglabel_class_enabled_notify ( void *obj, const char *lang )
 {
+  /* Intentionally empty: toggling "enabled" needs no side effect; the
+     property write is sufficient. Kept as a registered no-op hook point. */
 }
 
 CLASS_DOC(ratinglabel)

--- a/src/ratinglabels.c
+++ b/src/ratinglabels.c
@@ -612,11 +612,6 @@ ratinglabel_class_get_icon ( void *obj )
 static void
 ratinglabel_class_enabled_notify ( void *obj, const char *lang )
 {
-  int a=1;
-  ratinglabel_t *rl = obj;
-
-  if (rl->rl_enabled)
-    a++;
 }
 
 CLASS_DOC(ratinglabel)

